tests/test_add_start_node.py

```diff
@@ -0,0 +1,38 @@
+from selenium.webdriver.remote.webdriver import WebDriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver import ActionChains
+from selenium.common.exceptions import NoSuchElementException
+
+web = "https://machinist.flapflap.io/"
+
+
+def test_add_node(driver: WebDriver):
+
+    driver.get(web)
+    source = driver.find_element(By.XPATH, '//*[@id="__next"]/main/div[1]')
+    action = ActionChains(driver)
+    action.double_click(source).perform()
+
+    # check if node is added by verifying if there is text q1
+    inspect_val = driver.find_element(
+        By.XPATH, '//*[@id="__next"]/main/div[1]/div/div[2]/p'
+    ).text
+    true_val = "q1"
+
+    assert inspect_val == true_val
+
+
+def test_start_node(driver: WebDriver):
+    driver.get(web)
+    driver.find_element(By.XPATH, '//*[@id="__next"]/main/div[1]/div/div').click()
```

Comment:
same thing here I think you should give select an x,y coordinate before creating a node and not randomly click on the canvas